bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannAuth.java

```diff
@@ -209,6 +209,7 @@ private void getTokens() throws ViessmannAuthException {
         api.setTokenResponseDTO(tokenResponse);
         refreshToken = tokenResponse.refreshToken;
         api.setTokenExpiryDate(TimeUnit.SECONDS.toMillis(tokenResponse.expiresIn));
+        api.setRefreshTokenExpiryDate(TimeUnit.SECONDS.toMillis(REFRESH_TOKEN_EXPIRE));
```

Comment:
Is this correct? Is the token always re-generated when the endpoint is called? Or could it be that an old token is retrieved that expires early?

Reply:
This is correct. When the endpoint is called, a new refresh token will be generated.